src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,64 @@
 package core.basesyntax;
 
-public class WorkWithFile {
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
 
+public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(getCalculated(getFromFile(fromFileName)), toFileName);
+    }
+
+    private String getFromFile(String fromFileName) {
+        File myFile = new File(fromFileName);
+        StringBuilder dataFromFile = new StringBuilder();
+        try {
+            FileReader fileReader = new FileReader(myFile);
+            BufferedReader reader = new BufferedReader(fileReader);
+            String value = reader.readLine();
+            while (value != null) {
+                dataFromFile.append(value).append(" ");
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("", e);
+        }
+        return dataFromFile.toString();
+    }
+
+    private String getCalculated(String builder) {
+        StringBuilder calculation = new StringBuilder();
+        String[] splitData = builder.split(" ");
+        String[] fieldsToCalc = new String[]{"supply", "buy"};
+        int buy = 0;
+        int supply = 0;
+        for (String name : fieldsToCalc) {
+            int sumAmount = 0;
+            for (int i = 0; i < splitData.length; i++) {
+                String[] temp = splitData[i].split(",");
+                if (name.equals(temp[0])) {
+                    sumAmount += Integer.parseInt(temp[1]);
+                }
+            }
+            if (name.equals("buy")) {
+                buy = sumAmount;
+            } else {
+                supply = sumAmount;
+            }
+            calculation.append(name).append(",").append(sumAmount).append(System.lineSeparator());
+        }
```

Comment:
let's not use nested loops
```suggestion
        for (String line : splitData) {
                String[] parts = line.split(",");
                if ("buy".equals(parts[0])) {
                    buy += Integer.parseInt(parts[1]);
                } else {
                    supply += Integer.parseInt(parts[1]);
                }
            }
        }
```